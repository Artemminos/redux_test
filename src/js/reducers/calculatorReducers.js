const initialState = {
    output: 0,
    posts: []
};

const calculatorReducers = (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_INPUTS':
            return {...state, output: action.output};
        case 'SUBTRACT_INPUTS':
            return {...state, output: action.output};
        case 'ADD_POST':
            return {...state, posts: [...state.posts, action.post]};
        case 'REMOVE_POST':
            return {...state, posts: state.posts.filter(e=>e.id!==action.id)};
        default:
            return state;
    }
};

export default calculatorReducers;
