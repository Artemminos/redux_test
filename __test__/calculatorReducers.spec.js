import React from 'react'

import calculatorReducers from '../src/js/reducers/calculatorReducers'

describe('>>>R E D U C E R --- Test calculatorReducers', () => {
    it('+++ reducer for ADD_INPUT', () => {
        let state = {output: 100}
        state = calculatorReducers(state, {type: "ADD_INPUTS", output: 500})
        expect(state).toEqual({output: 500})
    });
    it('+++ reducer for SUBTRACT_INPUT', () => {
        let state = {output: 100}
        state = calculatorReducers(state, {type: "SUBTRACT_INPUTS", output: 50})
        expect(state).toEqual({output: 50})
    });

    it('+++ reducer for ADD_POST', () => {
        let state = {output: null, posts: []}
        state = calculatorReducers(state, {type: "ADD_POST", post: {name: 'newPost', id: 1}});
        expect(state.posts.length).toEqual(1);
    });

    it('+++ reducer remove POST', () => {
        let state = {output: null, posts: [{name: 'newPost', id: 1},{name: 'newPost', id: 2}]}
        state = calculatorReducers(state, {type: "REMOVE_POST", id:1});
        expect(state.posts.length).toEqual(1);
    });

    it('+++ reducer with shapshot', () => {
        expect(calculatorReducers(undefined, {type: 'default'})).toMatchSnapshot();
    });

    it('+++ reducer with shapshot', () => {
        const action = {
            type: 'ADD_INPUTS',
            output: 60,
        };
        expect(calculatorReducers(undefined, action)).toMatchSnapshot();
    });

});
//*******************************************************************************************************
