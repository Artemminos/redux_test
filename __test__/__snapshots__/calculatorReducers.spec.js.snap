exports[`>>>R E D U C E R --- Test calculatorReducers +++ reducer with shapshot 1`] = `
Object {
  "output": 0,
  "posts": Array [],
}
`;

exports[`>>>R E D U C E R --- Test calculatorReducers +++ reducer with shapshot 2`] = `
Object {
  "output": 60,
  "posts": Array [],
}
`;
